src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -493,22 +506,51 @@ private Result<?, UpdateError> validateAndHandleAddedTrip(
       return UpdateError.result(tripId, NO_START_DATE);
     }
 
-    // Check whether at least two stop updates exist
-    if (tripUpdate.getStopTimeUpdateCount() < 2) {
-      debug(tripId, "ADDED trip has fewer than two stops, skipping.");
+    final List<StopTimeUpdate> stopTimeUpdates = removeUnknownStops(tripUpdate, tripId);
+
+    var warnings = new ArrayList<UpdateSuccess.WarningType>(0);
+
+    if (stopTimeUpdates.size() < tripUpdate.getStopTimeUpdateCount()) {
+      warnings.add(UpdateSuccess.WarningType.UNKNOWN_STOPS_REMOVED_FROM_ADDED_TRIP);
+    }
+
+    // check if after filtering the stops we still have at least 2
+    if (stopTimeUpdates.size() < 2) {
+      debug(tripId, "ADDED trip has fewer than two known stops, skipping.");
       return UpdateError.result(tripId, TOO_FEW_STOPS);
     }
 
     // Check whether all stop times are available and all stops exist
-    final var stops = checkNewStopTimeUpdatesAndFindStops(tripId, tripUpdate);
+    final var stops = checkNewStopTimeUpdatesAndFindStops(tripId, stopTimeUpdates);
     if (stops == null) {
       return UpdateError.result(tripId, NO_VALID_STOPS);
     }
 
     //
     // Handle added trip
     //
-    return handleAddedTrip(tripUpdate, tripDescriptor, stops, tripId, serviceDate);
+    return handleAddedTrip(tripUpdate, stopTimeUpdates, tripDescriptor, stops, tripId, serviceDate)
+      .mapSuccess(s -> s.addWarnings(warnings));
```

Comment:
This feels a bit cumbersome for carrying the warnings

Reply:
Would you prefer if I passed in the warnings as an argument?